editoast/src/client/stdcm_search_env_commands.rs

```diff
@@ -0,0 +1,448 @@
+        .timetable_id(args.timetable_id)
+        .search_window_begin(begin)
+        .search_window_end(end)
+        .overwrite(conn)
+        .await?;
+
+    println!("✅ STDCM search environment set up successfully");
+    Ok(())
+}
+
+#[derive(Args, Debug)]
+#[command(
+    about,
+    long_about = "Set the current STDCM search env by copying most data from a scenario"
+)]
+pub struct SetSTDCMSearchEnvFromScenarioArgs {
+    pub scenario_id: i64,
+    #[arg(long)]
+    pub work_schedule_group_id: Option<i64>,
+    /// If omitted, set to the earliest train start time in the timetable
+    #[arg(long)]
+    pub search_window_begin: Option<NaiveDateTime>,
+    /// If omitted, set to the latest train start time in the timetable plus one day
+    #[arg(long)]
+    pub search_window_end: Option<NaiveDateTime>,
+}
+
+async fn set_stdcm_search_env_from_scenario(
+    args: SetSTDCMSearchEnvFromScenarioArgs,
+    conn: &mut DbConnection,
+) -> Result<(), Box<dyn Error + Send + Sync>> {
+    if let Some(work_schedule_group_id) = args.work_schedule_group_id {
+        exists_or_return!(WorkScheduleGroup, work_schedule_group_id, conn);
+    }
+
+    let scenario_option = Scenario::retrieve(conn, args.scenario_id).await?;
+
+    let scenario = scenario_option.ok_or_else(|| {
+        let error = CliError::new(
+            1,
+            format!("❌ Scenario not found, id: {0}", args.scenario_id),
+        );
+        Box::new(error)
+    })?;
+
+    let (begin, end) = resolve_search_window(
+        scenario.timetable_id,
+        args.search_window_begin,
+        args.search_window_end,
+        conn,
+    )
+    .await?;
+
+    StdcmSearchEnvironment::changeset()
+        .infra_id(scenario.infra_id)
+        .electrical_profile_set_id(scenario.electrical_profile_set_id)
+        .work_schedule_group_id(args.work_schedule_group_id)
+        .timetable_id(scenario.timetable_id)
+        .search_window_begin(begin)
+        .search_window_end(end)
+        .overwrite(conn)
+        .await?;
+
+    println!("✅ STDCM search environment set up successfully");
+    Ok(())
+}
+
+async fn resolve_search_window(
+    timetable_id: i64,
+    search_window_begin: Option<NaiveDateTime>,
+    search_window_end: Option<NaiveDateTime>,
+    conn: &mut DbConnection,
+) -> Result<(NaiveDateTime, NaiveDateTime), Box<dyn Error + Send + Sync>> {
+    let (begin, end) = if let (Some(begin), Some(end)) = (search_window_begin, search_window_end) {
+        (begin, end)
+    } else {
+        let start_times = Timetable::gather_start_times(timetable_id, conn).await?;
+
+        if start_times.is_empty() {
+            let error_msg =
+                "❌ Timetable specified contains no train. Please fully specify search window.";
+            return Err(Box::new(CliError::new(1, error_msg)));
+        }
+
+        let begin = search_window_begin
+            .unwrap_or_else(|| start_times.iter().min().expect("Vec.len() > 0").naive_utc());
```

Comment:
The `expect` error message might not help too much and end-user? How about something like "Needs at least one train in timetables".

Maybe you can even avoid entirely the `expect` by changing the above `if` by something like

```rust
let (Some(min), Some(max)) = (start_times.iter().min().naive_utc(), start_times.iter().max().naive_utc()) else {
    let error_msg =
        "❌ Timetable specified contains no train. Please fully specify search window.";
    return Err(Box::new(CliError::new(1, error_msg)));
}
```